cmake/TestHeaders.cmake

```diff
@@ -127,6 +127,7 @@ int main() { }
             ${ARGS_EXCLUDE_FROM_ALL}
             "${CMAKE_CURRENT_BINARY_DIR}/headers/${directory}/${filename}.cpp"
         )
+        target_include_directories(test.header.${target} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include)
```

Comment:
The point of the header tests is to check that all headers include all the files they need to compile, so that if a user only wants to use, e.g., `range/v3/utility/static_const.hpp`, they can just include that file and their program will compile fine (because that header includes all the headers it needs to be self-suficient).

Does that make sense? 

The way that is tested is by auto-generating programs like this for each header in the library and checking that they compile:

```c++
#include <range/v3/utility/static_const.hpp>
int main() {}
```